TeamCode/src/main/java/org/firstinspires/ftc/teamcode/ultimategoal/Robot.java

```diff
@@ -70,12 +70,19 @@ public void update() {
                 } catch (Exception e) {
                     Log.d("Module", "Module couldn't update: " + module.getName());
                 }
-                if (WILL_FILE_DUMP) {
-                    module.fileDump();
-                }
             }
         }
         telemetryDump.update();
+
+        if(isStopRequested()) {
+            this.cleanUp();
+        }
+    }
+
+    private void cleanUp() {
+        this.fileDump.writeFilesToDevice();
+        for(Module module : this.modules)
+            module.close();
```

Comment:
What I meant was to close the ModuleExecutor thread, I don't really think we need to worry about the modules themselves?